openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -334,20 +337,82 @@ export function initAuthorMultiInputAutocomplete() {
                 minChars: 2,
                 max: 11,
                 matchSubset: false,
-                autoFill: false,
+                autoFill: true,
                 formatItem: render_author_autocomplete_item
             });
     });
 }
 
+export function initSubjectsAutocomplete() {
+    function splitField(val) {
+        const re = /"?((?<=")[^"]+|(?<!")[^,]+)"?[, "]*/g;
+        return Array.from(val.matchAll(re), (m) => m[1]);
+    }
+
+    function joinField(vals) {
+        const escaped = vals.map(val => (val.includes(',')) ? `"${val}"` : val);
+        return escaped.join(', ');
+    }
+
+    getJqueryElements('.multi-input-autocomplete--subjects').forEach(jqueryElement => {
+        /* Values in the html passed from Python code */
+        const dataConfig = JSON.parse(jqueryElement[0].dataset.config);
+        jqueryElement.setup_multi_input_autocomplete(
```

Comment:
Hmmm... so `multi-input-autocomplete` is used for like the authors or languages; it expects multiple `<input>` tags. Its working fine here because we happen to be using a `<textarea>` tag, but that's a bit subtle.

I think create a new `$.fn.setup_csv_autocomplete` function inside autocomplete.js that calls `setup_autocomplete`, which does everything except the input stuff. Then you can move all these functions there. Then here, we just call `.setup_csv_autocomplete({...})`. In theory we could use this same logic for eg publishers in the future.

Reply:
Then also change the class name from `multi-input-autocomplete--subjects` to `csv-autocomplete--subjects`

Reply:
Turns out the textareas do need IDs, or the associated label elements get very mad.

Otherwise, refactored as you suggested. The improved reusability is a nice bonus.